AGXUnity/BrickUnity/Factories/ConstraintFactory.cs

```diff
@@ -37,125 +47,339 @@ public static void SetControllers(this AGXUnity.Constraint constraint, B_Connect
+      }
     }
 
-    public static void SetComplianceAndDamping(this AGXUnity.Constraint constraint, B_Interaction b_interaction)
+    /// <summary>
+    /// Check if a Brick interaction data value (compliance or damping) is default (not manually set by the user)
+    /// </summary>
+    /// <param name="data">The Brick interaction data</param>
+    /// <param name="dof">The degree of freedom for which to check</param>
+    /// <returns>True if the interaction data value is default</returns>
+    public static bool GetValueIsDefault(this B_Mechanics.InteractionData6D data, AU_Constraint.TranslationalDof dof)
     {
+      switch (dof)
+      {
+        case AU_Constraint.TranslationalDof.X:
+          return data._alongNormalIsDefault;
+        case AU_Constraint.TranslationalDof.Y:
+          return data._alongCrossIsDefault;
+        case AU_Constraint.TranslationalDof.Z:
+          return data._alongTangentIsDefault;
+        default:
+          return data._alongNormalIsDefault && data._alongCrossIsDefault && data._alongTangentIsDefault;
+      }
+    }
+
+    /// <summary>
+    /// Set damping of an AGXUnity constraint from Brick, for a specific degree of freedom
+    /// </summary>
+    /// <param name="constraint">The AGXUnity constraint on which to set the damping</param>
+    /// <param name="damping">The Brick damping to set</param>
+    /// <param name="dof">The degree of freedom for which to set the damping</param>
+    /// <param name="overwriteIfDefault">Set to true to overwrite the AGXUnity constraint's values even if the Brick values are default</param>
+    public static void SetBrickDamping(this AU_Constraint constraint,
+                                       B_Damping damping,
+                                       AU_Constraint.RotationalDof dof,
+                                       bool overwriteIfDefault)
+    {
+      if (overwriteIfDefault || !damping.GetValueIsDefault(dof))
+        constraint.SetDamping(damping.GetValue(dof) * constraint.GetDamping(dof), dof);
+    }
+
+    /// <summary>
+    /// Set damping of an AGXUnity constraint from Brick, for a specific degree of freedom
+    /// </summary>
+    /// <param name="constraint">The AGXUnity constraint on which to set the damping</param>
+    /// <param name="damping">The Brick damping to set</param>
+    /// <param name="dof">The degree of freedom for which to set the damping</param>
+    /// <param name="overwriteIfDefault">Set to true to overwrite the AGXUnity constraint's values even if the Brick values are default</param>
+    public static void SetBrickDamping(this AU_Constraint constraint,
+                                       B_Damping damping,
+                                       AU_Constraint.TranslationalDof dof,
+                                       bool overwriteIfDefault)
+    {
+      if (overwriteIfDefault || !damping.GetValueIsDefault(dof))
+        constraint.SetDamping(damping.GetValue(dof) * constraint.GetDamping(dof), dof);
```

Comment:
Same with the damping here

Reply:
Fixed in eba0b71eaf9fe84c9c7f3dcf2fc580404e68e092